src/utils/holdPageUtils.ts

```diff
@@ -0,0 +1,85 @@
+import type {
+  EDDRequestParams,
+  EDDFormValidatedField,
+  EDDPageStatus,
+  EDDStatusMessage,
+} from "../types/holdPageTypes"
+
+import { EMAIL_REGEX } from "../config/constants"
+
+export const initialEDDFormState: EDDRequestParams = {
+  email: "",
+  startingNumber: "",
+  endingNumber: "",
+  chapter: "",
+  publicationDate: "",
+  volume: "",
+  issue: "",
+  itemId: "",
+  patronId: "",
+  source: "",
+  author: "",
+  notes: "",
+  pickupLocation: "edd",
+}
+
+export const EDDPageStatusMessages: Record<EDDPageStatus, EDDStatusMessage> = {
+  failed: {
+    heading: "Request failed",
+    message: "We were unable to process your request at this time.",
+  },
+  unavailable: {
+    heading: "Electronic delivery unavailable",
+    message:
+      "Electronic delivery options for this item are currently unavailable.",
+  },
+  invalid: {
+    message: "Some fields contain errors. Please correct and submit again.",
+  },
+}
+
+// Initial state for invalid fields in the EDD form to keep track of the first invalid field for focus on submit
+export const initialEDDInvalidFields: EDDFormValidatedField[] = [
+  { key: "email", isInvalid: false },
+  { key: "startingNumber", isInvalid: false },
+  { key: "endingNumber", isInvalid: false },
+  { key: "chapter", isInvalid: false },
+]
+
+// Gets updated invalidFields to set them in state based on the inputted field name and value
+export const getUpdatedInvalidFields = (
+  fieldName: string,
+  fieldValue: string,
+  prevInvalidFields: EDDFormValidatedField[]
+): EDDFormValidatedField[] => {
+  return prevInvalidFields.map((field) => {
+    if (field.key === fieldName) {
+      switch (field.key) {
+        // Validate email field
+        case "email":
+          return {
+            key: "email",
+            isInvalid: !fieldValue.length || !EMAIL_REGEX.test(fieldValue),
```

Comment:
I forgot already, but is this similar or the same validation as the My Account email test?

Reply:
I don't know that Emma's work is referencing this constant yet, but this constant is based on Emma's validation regex.